pkg/controllers/endpointslice_plan.go

```diff
@@ -0,0 +1,176 @@
+package controllers
+
+import (
+	"github.com/aws/aws-cloud-map-mcs-controller-for-k8s/pkg/model"
+	v1 "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1beta1"
+)
+
+const defaultMaxEndpointsPerSlice = 100
+
+type EndpointSliceChanges struct {
+	// Create: List of EndpointSlices that need to be created
+	Create []*discovery.EndpointSlice
+	// Update: List of EndpointSlices that need to be updated
+	Update []*discovery.EndpointSlice
+	// Delete: List of EndpointSlices that need to be deleted
+	Delete []*discovery.EndpointSlice
+	// Unmodified: List of EndpointSlices that do not need to be changed
+	Unmodified []*discovery.EndpointSlice
+}
+
+type EndpointSlicePlan struct {
+	// maxEndpointsPerSlice defaults to 100
+	maxEndpointsPerSlice int
+
+	// Service to reconcile endpoints in
+	Service *v1.Service
+
+	// ServiceImportName name used to create new EndpointSlices
+	ServiceImportName string
+
+	// Current EndpontSlices
+	Current []discovery.EndpointSlice
```

Comment:
Could be optimal to use list of pointers here so we don't copy the complete list.

Reply:
That's fair.